### PR TITLE
Add venue ID to venue functions in the backend

### PR DESCRIPTION
**Pull request by colinhowe**
This creates a single unique and non-changing way of finding a venue in the database.

This closes https://github.com/sparkletown/internal-sparkle-issues/issues/1522

This also removes some older functions as updating them to support world ID made little sense.

### CODE REVIEW

File: functions/src/utils/venue.js [hunk 1]
```diff
@@ -3,18 +3,11 @@ const { VenueIdSchema } = require("../types/venue");
 const VENUE_ID_REGEX = /[a-z0-9_]{1,250}/;
 const INVALID_SLUG_CHARS_REGEX = /[^a-zA-Z0-9]/g;
 
-/**
- * @deprecated Use generateSlug instead.
- */
-const getVenueId = (name) =>
-  name.replace(INVALID_SLUG_CHARS_REGEX, "").toLowerCase();
```
Comment (colinhowe) [inline functions/src/utils/venue.js:10]: This function was misleading as slug is not always going to be the ID of a venue.

File: functions/venue.js [hunk 2]
```diff
@@ -49,27 +49,6 @@ const VenueTemplate = {
...
-  VenueTemplate.jazzbar,
-  VenueTemplate.partymap,
-  VenueTemplate.animatemap,
-  VenueTemplate.performancevenue,
-  VenueTemplate.themecamp,
-  VenueTemplate.viewingwindow,
-  VenueTemplate.zoomroom,
-  VenueTemplate.screeningroom,
-];
-
```
Comment (colinhowe) [inline functions/venue.js:72]: These all became unused after I deleted the v1 create venue function.

File: functions/venue.js [hunk 3]
```diff
@@ -180,117 +159,6 @@ const checkIfUserHasVoted = async (venueId, pollId, userId) => {
...
-      venueData.roomVisibility = data.roomVisibility;
-      break;
-
-    default:
-      break;
-  }
-
-  return venueData;
-};
-
```
Comment (colinhowe) [inline functions/venue.js:293]: Removed this function as adding support for a proper venue ID to be passed in didn't make any sense.

File: functions/venue.js [hunk 4]
```diff
@@ -517,27 +385,6 @@ exports.removeVenueOwner = functions.https.onCall(async (data, context) => {
...
-  const venueId = getVenueId(data.name);
-  const venueRef = admin.firestore().collection("venues").doc(venueId);
-
-  batch.set(venueRef, venueData);
-  initializeVenueChatMessagesCounter(venueRef, batch);
-
-  await batch.commit();
-
-  return venueData;
-});
```
Comment (colinhowe) [inline functions/venue.js:539]: Deleting this function was easier than trying to shoe-horn in the venue ID.

File: src/api/admin.ts [hunk 5]
```diff
@@ -289,6 +295,20 @@ const createFirestoreVenueInput_v2 = async (
...
+const createFirestoreVenueInput_v2 = async (
+  input: VenueInput_v2,
+  user: firebase.UserInfo
+) => {
+  // We temporarily cast the result to unknown so that we can cast to the
+  // same type with the ID added, then we add the missing property.
+  const result = ((await createFirestoreVenueInputWithoutId_v2(
+    input,
+    user
+  )) as unknown) as FirestoreVenueInput_v2;
```
Comment (colinhowe) [inline src/api/admin.ts:307]: I don't like this. I couldn't think of a cleaner way to do this though.
The return from `createFirestoreVenueInputWithoutId_v2` does not have an ID. We want to add that ID property on to both the object and the type of that object. Open to ideas on cleaner ways to do this.

File: src/api/admin.ts [hunk 6]
```diff
@@ -300,10 +320,10 @@ export const createVenue = async (
 };
 
 export const createVenue_v2 = async (
-  input: WithWorldId<VenueInput_v2>,
+  input: WithWorldId<Omit<VenueInput_v2, "id">>,
```
[the remark this answers is not in the document]
Comment (colinhowe) [reply src/api/admin.ts:323]: I want the default for "doing things with venues" to require a venue ID. Then the special case is venue creation which doesn't need (or want) an ID.

**Review by denisdimitrov — APPROVED** [review]
LGTM!

**Pull request merged**